### PR TITLE
Update RC1 migration docs

### PR DESCRIPTION
**Pull request by mslukebo**

### CODE REVIEW

File: documentation/Migration/Update-1.0.0-rc1.md [hunk 1]
```diff
@@ -1,15 +1,86 @@
...
+  - [Plugin Configurations](#plugin-configurations)
+- [Suggested Changes](#suggested-changes)
+  - [Metadata Tables](#metadata-tables-1)
+
+# Conceptual Changes
+
+## Table Discovery
+
+Prior to this release candidate, any tables defined in a processing source's assembly were associated with
+that processing source. The SDK runtime ensured that tables that do not provide a static build method are built
```
Comment (jmaxson-ms) [inline documentation/Migration/Update-1.0.0-rc1.md:56]: maybe "The SDK base classes ensured that tables..."? I don't think it was actually the "runtime".

File: documentation/Migration/Update-1.0.0-rc1.md [hunk 2]
```diff
@@ -1,15 +1,86 @@
...
+by passing their `TableDescriptor`s to the processing source's `CustomDataProcessor`'s `BuiltTable` method.
```
Comment (jmaxson-ms) [inline documentation/Migration/Update-1.0.0-rc1.md:57]: >BuiltTable [](http://example.com/codeflow?start=88&length=10)

BuiltTable -> BuildTable

File: documentation/Migration/Update-1.0.0-rc1.md (hunk 1 above)
Comment (mslukebo) [reply documentation/Migration/Update-1.0.0-rc1.md:56]: I actually think nothing was enforcing this... at most the Engine was

Comment (mslukebo) [reply documentation/Migration/Update-1.0.0-rc1.md:56]: maybe I can say "The SDK Engine (and most SDK drivers)"?

Comment (jmaxson-ms) [reply documentation/Migration/Update-1.0.0-rc1.md:56]: Maybe don't even mention it. Just note the transition from additional tables to the provider?

File: documentation/Migration/Update-1.0.0-rc1.md [hunk 3]
```diff
@@ -261,7 +370,14 @@ There are some changes made to interfaces that generally don't need to be implem
...
+- The following members have been removed:
+
+    * method `ICustomDataProcessor.BuildMetadataTables`
+
+- The following members have been added:
+    - method `TryEnableTable`. This is similar to `EnableTable`, but catches most exceptions and returns a `bool`.
+
+- `EnableTable` and `ProcessAsync` may now throw an `InvalidOperationException` if `ProcessAsync` has already been called.
 
 ### IDataProcessorExtensibilitySupport
```
Comment (jmaxson-ms) [inline documentation/Migration/Update-1.0.0-rc1.md:382]: >IDataProcessorExtensibilitySupport [](http://example.com/codeflow?start=4&length=34)

IDataProcessorExtensibilitySupport has now been removed as it was only around to support internal tables.

File: src/Microsoft.Performance.SDK/Processing/IProcessingSource.cs [hunk 4]
```diff
@@ -25,14 +25,14 @@ public enum SerializationSource
     public interface IProcessingSource
     {
         /// <summary>
-        ///     Gets the collection of tables exposed by this <see cref="IProcessingSource"/>.
-        ///     These are the data tables for exposing the processed data.
+        ///     Gets the collection of tables exposed by this <see cref="IProcessingSource"/>
+	///     which are not marked as <see cref="TableDescriptor.IsMetadataTable"/>
```
Comment (jmaxson-ms) [inline src/Microsoft.Performance.SDK/Processing/IProcessingSource.cs:29]: >	///  [](http://example.com/codeflow?start=0&length=5)

nit: in CodeFlow, it looks like indentation is off for this, and below

File: documentation/Migration/Update-1.0.0-rc1.md (hunk 2 above)
Comment (mslukebo) [reply documentation/Migration/Update-1.0.0-rc1.md:57]: removed this sentence

File: src/Microsoft.Performance.SDK/Processing/IProcessingSource.cs (hunk 4 above)
Comment (mslukebo) [reply src/Microsoft.Performance.SDK/Processing/IProcessingSource.cs:29]: yeah it is. edited outside of visual studio :/

File: documentation/Migration/Update-1.0.0-rc1.md (hunk 3 above)
Comment (mslukebo) [reply documentation/Migration/Update-1.0.0-rc1.md:382]: thanks!

**Review by jmaxson-ms — APPROVED** [review]

**Review by zachnew-msft — APPROVED** [review]

**Pull request merged**